xchange-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioAdapters.java

```diff
@@ -71,4 +78,53 @@ private static BigDecimal adaptAmount(BigDecimal amount) {
         }
         return status;
     }
+    
+    private static List<LimitOrder> updateOrInsertQuantityForPrice(List<LimitOrder> existingOrders, 
+    		List<List<BigDecimal>> updatedQuantitiesPerPrice,
+    		OrderType orderType,
+    		Date timestamp,
+    		CurrencyPair currencyPair,
+    		String id) {
+    	Collections.sort(existingOrders);
+    	for (List<BigDecimal> updatedQuantityPrice : updatedQuantitiesPerPrice) {
+    		if (updatedQuantityPrice.size() != 2) {
+    			throw new IllegalArgumentException("Expected price quantity list to contain exactly two big decimals");
+    		}
+    		BigDecimal price = updatedQuantityPrice.get(0);
+    		BigDecimal quantity = updatedQuantityPrice.get(1);
+    		LimitOrder orderForUpdatedQuantityPrice = new LimitOrder(orderType, quantity, currencyPair, id, timestamp, price);
+    		int insertIndex = Collections.binarySearch(existingOrders, orderForUpdatedQuantityPrice);
+    		if (insertIndex == existingOrders.size()) {
+    			existingOrders.add(orderForUpdatedQuantityPrice);
+    		}
+    		else if (insertIndex < 0)
+    		{
+    			existingOrders.add(0, orderForUpdatedQuantityPrice);
+    		}
+    		else if (existingOrders.get(insertIndex).compareTo(orderForUpdatedQuantityPrice) == 0) 
+    		{
+    			existingOrders.set(insertIndex, orderForUpdatedQuantityPrice);
+    		} 
+    		else 
+    		{
+    			existingOrders.add(insertIndex, orderForUpdatedQuantityPrice);
+    		}
+    	}
+    	return existingOrders;
+    }
+    
+    protected static CurrencyPair adaptCurrencyPairString(String currencyPairString) {
```

Comment:
I'd be very surprised if XChange doesn't already provide a method for this!

Reply:
They do but it is private :( I copied their implementation exactly with a note in the code about the duplication. I can submit a PR to them to make the function protected if that would be useful.